src/Symfony/Component/Form/Extension/Core/Type/ButtonType.php

```diff
@@ -43,8 +43,6 @@ public function configureOptions(OptionsResolver $resolver)
     {
         parent::configureOptions($resolver);
 
-        $resolver->setDefaults([
-            'auto_initialize' => false,
-        ]);
+        $resolver->setDefault('auto_initialize', false);
```

Comment:
I would keep the former version.
If one will add a new option in the future the diff will be much more readable, but it’s a small change, just per personal opinion though

Reply:
I don't agree, the core should be optimised here, best practices for user applications is not the same as writing core code (DX should not apply in core). IMHO all core type should chain `setDefault` calls to prevent useless `foreach` loop.

Reply:
Fair enough 👍🏻